components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/WriterGroupMessageSource.cs

```diff
@@ -30,11 +30,56 @@ public class WriterGroupMessageTrigger : IMessageTrigger, IDisposable {
         public int NumberOfConnectionRetries => _subscriptions?.FirstOrDefault()?.
             Subscription?.NumberOfConnectionRetries ?? 0;
 
+        private IDictionary<DateTime, int> _valueChanges = new Dictionary<DateTime, int>(100);
```

Comment:
>new Dictionary<DateTime, int>(100) [](http://example.com/codeflow?start=59&length=34)

idea: use a histogram array (ringbuffer which counts e.g. 10 s blocks). Then no need to manage dictionary content

Reply:
good idea, changed implementation to ring buffer